conseil-lorre/src/test/scala/tech/cryptonomic/conseil/indexer/ethereum/EthereumOperationsTest.scala

```diff
@@ -0,0 +1,143 @@
+      }
+
+      "run indexer for the all blocks" in new EthereumOperationsStubs {
+        // Bitcoin operations with mocked loadBlocksWithTransactions to test only loadBlocks.
+        val ethereumOperationsStub = new EthereumOperationsMock {
+          override def loadBlocksWithTransactions(range: Range.Inclusive) =
+            ethereumOperationsMock.loadBlocksWithTransactions(range)
+        }
+
+        // check if the method gets the latest block at the beginning
+        (ethereumPersistenceMock.getLatestIndexedBlock _) expects ()
+        (txMock.transact[Option[Tables.BlocksRow]] _) expects (*) returning (IO(None))
+
+        // mock Bitcoin client calls
+        (ethereumClientMock.getMostRecentBlockNumber _) expects () returning (Stream("0xA").covary[IO])
+
+        // test if method runs indexer with the correct range of the blocks
+        (ethereumOperationsMock.loadBlocksWithTransactions _) expects (1 to 10) returning (Stream.empty)
+
+        // run
+        ethereumOperationsStub.loadBlocksAndLogs(Everything).compile.drain.unsafeRunSync()
+      }
+
+      "run indexer for the custom blocks range" in new EthereumOperationsStubs {
+        // Bitcoin operations with mocked loadBlocksWithTransactions to test only loadBlocks.
+        val ethereumOperationsStub = new EthereumOperationsMock {
+          override def loadBlocksWithTransactions(range: Range.Inclusive) =
+            ethereumOperationsMock.loadBlocksWithTransactions(range)
+        }
+
+        // check if the method gets the latest block at the beginning
+        (ethereumPersistenceMock.getLatestIndexedBlock _) expects ()
+        (txMock.transact[Option[Tables.BlocksRow]] _) expects (*) returning (IO(Some(blockRow.copy(number = 3))))
+
+        // mock Bitcoin client calls
+        (ethereumClientMock.getMostRecentBlockNumber _) expects () returning (Stream("0xA").covary[IO])
+
+        // test if method runs indexer with the correct range of the blocks
+        (ethereumOperationsMock.loadBlocksWithTransactions _) expects (7 to 10) returning (Stream.empty)
+
+        // run
+        ethereumOperationsStub.loadBlocksAndLogs(Custom(3)).compile.drain.unsafeRunSync()
+      }
+    }
+
+  /**
+    * Stubs that can help to provide tests for the [[EthereumOperations]].
+    *
+    * Usage example:
+    *
+    * {{{
+    *   "test name" in new EthereumOperationsStubs {
+    *     // ethereumOperationsStub is available in the current scope
+    *   }
+    * }}}
+    */
+  trait EthereumOperationsStubs {
+
+    val batchConfig = EthereumBatchFetchConfiguration(
+      indexerThreadsCount = 1,
+      httpFetchThreadsCount = 1,
+      blocksBatchSize = 1,
+      transactionsBatchSize = 1,
+      logsBatchSize = 1
+    )
+
+    class EthereumClientMock(rpcClient: RpcClient[IO]) extends EthereumClient[IO](rpcClient)
+    class EthereumPersistenceMock extends EthereumPersistence[IO]
+
+    val ethereumClientMock = mock[EthereumClientMock]
```

Comment:
I fail to see the value of having a mock over a mock instead of a mock of the real thing? 🤔